src/twisted/web/_newclient.py

```diff
@@ -813,7 +813,14 @@ def writeTo(self, transport):
             else:
                 return self._writeToContentLength(transport)
         else:
-            self._writeHeaders(transport, None)
+            # If the method semantics anticipate a body, include a
```

Comment:
I'd like to see `_writeToContentLength` renamed `_writeToContentLengthWithBodyProducer` or something and this code lifted into a new method called `_writeToContentLengthWithoutBody`.  I think it's confusing to have a `_writeToContentLength` method that doesn't always write the content length.